dexcom/fetch/translate.go

```diff
@@ -40,12 +40,31 @@ const MinimumOffsets = (-12 * time.Hour) / OffsetDuration // Minimum time zone o
 const DailyDuration = 24 * time.Hour
 const DailyOffsets = DailyDuration / OffsetDuration
 
-func translateTime(systemTime *dexcom.Time, displayTime *dexcom.Time, datum *dataTypes.Base) {
+// Expectations:
+// - systemTime must not be nil
+// - displayTime can be nil (systemTime used, if so)
+// - datum must not be nil
+func TranslateTime(systemTime *dexcom.Time, displayTime *dexcom.Time, datum *dataTypes.Base) {
```

Comment:
Not sure I agree with your interpretation. It is in the language specification and thus a feature to be used. What it really means is "modify the thing in place" which is what I want to do here. I don't want to create a copy (and all that entails - deep or shallow copy, ...).

Reply:
> It is in the language specification and thus a feature to be used. 

I didn't say it wasn't, in fact I pointed out a place where it is used. Certainly you can understand that just because a feature exists, doesn't mean it should be used. Modifying passed arguments is certainly possible, but it's relatively rare, and especially given the name of the function, I don't think it's at all expected.

> What it really means is "modify the thing in place" which is what I want to do here.

Yes, so, since you don't want to make a copy/clone (which is totally fine), would you be opposed to a name that makes it obvious that `datum` is being modified, something like UpdateDatumTimes perhaps?

I wonder if this function wouldn't make more sense as a method on dataTypes.Base, a call like `datum.UpdateFromTimes(systemTime *dexcom.Time, displayTime *dexcom.Time)` is pretty clear in what it does.

Reply:
I suppose I could change the name if that will clear this up for you. Adding the function to the `dataTypes.Base` type would be mixing in the `dexcom` package into a whole pile of code where it absolutely does not belong. (If Golang had a way to locally extend a struct, then maybe, but it does not.)

Reply:
> Adding the function to the dataTypes.Base type would be mixing in the dexcom package ...

This would be my sentiment with regards to `datum.UpdateFromTimes(`. 

It might not be perfect but we are using this pattern  see with the `func TranslateTime(` often already so its nothing different

Reply:
Yeah, re-reading this today, I realize I was being a doofus, sorry.